react/ManualPrice.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { FunctionComponent, Fragment, useState } from 'react'
+import { FormattedMessage, useIntl } from 'react-intl'
+import { FormattedPrice } from 'vtex.formatted-price'
+import { Button, ButtonPlain, InputCurrency, Modal, Tag } from 'vtex.styleguide'
+import { useRuntime } from 'vtex.render-runtime'
+
+import { useItemContext } from './ItemContext'
+import { getFormattedPrice } from './utils/price'
+
+const ManualPrice: FunctionComponent = () => {
+  const {
+    item,
+    itemIndex,
+    shouldAllowManualPrice,
+    onSetManualPrice,
+  } = useItemContext()
+  const [isModalOpen, setIsModalOpen] = useState(false)
+  const [manualPrice, setManualPrice] = useState(item.manualPrice)
+  const {
+    culture: { currency, locale },
+  } = useRuntime()
+  const intl = useIntl()
+  const priceChanged =
+    item.sellingPrice === item.manualPrice && item.sellingPrice !== item.price
+
+  const submitManualPrice = (evt: React.FormEvent<HTMLFormElement>) => {
+    evt.preventDefault()
+
+    if (manualPrice != null && !isNaN(manualPrice)) {
```

Comment:
what do you want to check here if `!isNaN(manualPrice)`?

either way, you should use `Number.isNaN` instead of the new ES5 global `isNaN` because it will check if the number is equal to `NaN` instead of checking if `typeof manualPrice !== 'number'`

```suggestion
    if (manualPrice != null && !Number.isNaN(manualPrice)) {
```

Reply:
I did this because if you try to click on the apply button with a empty input, the `manualPrice` value would be `NaN`. And so, the first part of this check (`manual-price != null`) wouldn't catch this. Let me know if you have a better way of doing this check!

Reply:
I tried to submit the manual price with the empty input but the price was `null`. What are the steps for it to become `NaN`?

Reply:
I figured it out! 😅 but from what I can see, it is still possible to change the manual price to NaN

Reply:
Yeah, I found it testing your suggestions. 😅 I'm already on it.

Reply:
Could you give it a try again?